### DISCUSSION TITLE
Clearing setup after code change

### DISCUSSION THREAD

**Discussion by andreped** [Project-MONAI/MONAILabel#757] [Q&A] [answered]
In a project, we have been using MONAI Label to annotate CT data and train segmentation models. However, we have found a strange behaviour which I don't know how to solve. We have been using the [DeepEdit](https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/deepedit.py) sample app (in radiology).

If I run the app directly, everything works fine. However, if I change the model, it fails, as I believe we need to remove the generated old model and model files first. After doing so, the training works fine but the loss is very strange. It is close to zero (it seemed to train fine the very first time). Is there something else I need to remove? Something stored in cache maybe? Is there a simple way to reset all previous trainings and start from scratch (without the need of re-downloaing the sample app, incorporating all my previous modifications, before starting training all over)? Seems very strange that this at all is necessary. Any ideas?

**Comment by SachidanandAlle** [comment, marked as answer]
@diazandr3s can help and provide more insights on deepedit related..

**Reply by andreped** [reply]
We found out what the issue was. We wanted to use the [segmentation_spleen](https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/segmentation_spleen.py) app instead, which produced segmentations that made more sense to us.

We had some other issues with training, where I believe the background class was added into the loss computation. Why not just use Dice of the class(es) of interest? We often neglect the background class when building and evaluating models (as the background, being so common, typically inflates the loss/metrics).

**Reply by SachidanandAlle** [reply]
Correct.. background label is bit confusing for deepedit.. thats one reason we have simplified segmentation/spleen example for users to start with..

**Reply by andreped** [reply]
OK, but then I guess we just need to make some modifications to the segmentation/spleen example(s) to make them work for our use case. Thanks!

**Reply by SachidanandAlle** [reply]
yes.. all the models are purely examples.. just for reference for users.. 
the models are trained on very few images from public dataset plus simple pre-transforms are mostly of tightly coupled to the type of images we used for training... so, we can fairly assume they won't fit all use-cases..

**Reply by diazandr3s** [reply]
Sorry, I just saw this discussion. 
Yes, the reason we have a background label in DeepEdit is that we specifically simulate background clicks. 
In any case, I also think we need to simplify and make the models (segmentation and DeepEdit) homogeneous.
Thanks for opening this discussion, @andreped

**Reply by andreped** [reply]
I also think that is a good idea, @diazandr3s.

I think for MONAI Label to be more relevant to clinicians and people who necessarily don't have a technical background, the segmentation app needs to be a little bit more plug-and-play, without the need to adjust the source code itself.

What we normally do when we setup training pipelines, is that we define hyperparameters, such as batch size, number of epochs, learning rate, and which network to use, **in a configuration file** (e.g., .ini). Being able to choose hyperparameters through the GUI would be a great feature. Even one that I would use :] It is also necessary for non-technical users.

**Reply by diazandr3s** [reply]
Agree! Thanks again for the feedback :)
